onnxruntime/core/providers/cpu/tensor/unique.cc

```diff
@@ -81,13 +81,13 @@ Status Unique::Compute(OpKernelContext* context) const {
   auto data_type = input.DataType();
 
   // arbitrary set of types to support initially
-  if (data_type == DataTypeImpl::GetType<float>())
+  if (utils::IsPrimDataType<float>(data_type))
     status = ComputeImpl<float>(*context);
```

Comment:
What's the rationale between choosing the dispatch on types in some places and an if/else if in others?

Reply:
I am going to add a comment to utils.h as follows:
1) If you need to handle all of the primitive datatypes, the best choice would be macros DispatchOnTensorType or DispatchOnTensorTypeWithReturn. Use inline wrappers so the function can be called as function<T>
2) if you have a few types, use utils::IsPrimaryDataType<T>() for if/else clause
3) For something in between, we suggest to use CallDispatcher pattern.

---
In reply to: [341932239](https://github.com/microsoft/onnxruntime/pull/2307#discussion_r341932239) [](ancestors = 341932239)